### PR TITLE
Weed rework

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

This gives xenos 5 different weeds:

- The speedy weed, which will improve xeno speed when stepping on it
- The sticky weed, which slow down marines 
- The healing weed, which you will have to go to rest
- The toxin weed, marines stepping on it takes toxins damage
- The fire weed, this makes resin structure on those immune to fire (not sure about this though, if you have ideas on weeds feel free to propose)

All of these weeds have only one effect, but they are overall better (aka, healing weed heals a tad better than the current basic weed, speedy weed gives slightly better boost etc)

Weeds also are decaying faster

Removes sticky from the game, so sticky spit will now longer produces it but will have better stagger on impact

## Why It's Good For The Game

I hope this add depth to xeno gameplay, especially drone/hivelord one.

Overall, this is a xeno nerf i think, because i was conservative with the values, but i don't know what will xeno be able to do with this

## Changelog
:cl:
add: Weeds type, xeno have now access to five different weeds. Resting weed, speed weed, sticky weed, toxin weed and fire resistance weed. You won't be able to regen health without being on a resting weed
/:cl:

### CODE REVIEW

**Comment by Surrealaser** [comment]
I don't know about this.

Could be cool, but needs one hell of a playtest.

**Comment by RenGusta** [comment]
Should probably reduce the basic effect of weeds not just have these as weed+

File: code/controllers/subsystem/weeds.dm [hunk 1]
```diff
@@ -8,6 +8,8 @@ SUBSYSTEM_DEF(weeds)
 	var/list/creating = list()
 	var/list/pending = list()
 	var/list/currentrun
+	/// The amount of time a weed will try to spawn on a given tile
+	var/list/attempt = list()
```
Comment (BraveMole) [inline code/controllers/subsystem/weeds.dm:12]: I discovered that some weeds could indefinitly try to spawn, until their node is destroyed

File: code/__DEFINES/_math.dm [hunk 2]
```diff
@@ -7,7 +7,8 @@
 
 #define CARDINAL_DIRS list(1,2,4,8)
 #define CARDINAL_ALL_DIRS list(1,2,4,5,6,8,9,10)
-#define cheap_hypotenuse(Ax,Ay,Bx,By) (sqrt(abs(Ax - Bx)**2 + abs(Ay - By)**2)) //A squared + B squared = C squared
+#define get_dist_euclide_square(A, B) (A && B ? A.z == B.z ? (A.x - B.x)**2 + (A.y - B.y)**2 : INFINITY : INFINITY)
+#define get_dist_euclide(A, B) (sqrt(get_dist_euclide_square(A, B)))
```
Comment (BraveMole) [inline code/__DEFINES/_math.dm:11]: we should use these everywhere btw

File: code/controllers/subsystem/weeds.dm [hunk 3]
```diff
@@ -17,89 +19,90 @@ SUBSYSTEM_DEF(weeds)
...
 
 		if (!T.is_weedable())
 			pending -= T
+			attempt -= T
 			continue
 
 		for(var/direction in GLOB.cardinals)
 			var/turf/AdjT = get_step(T, direction)
-			if (!(AdjT in N.node_turfs)) // only count our weed graph as eligble
-				continue
```
Comment (BraveMole) [inline code/controllers/subsystem/weeds.dm:43]: this created jittery in the creation of the weeds. It's not really needed with the new get_dist system

File: code/game/objects/structures/xeno.dm [hunk 4]
```diff
@@ -104,18 +105,7 @@
...
-// Praetorian Sticky Resin spit uses this.
-/obj/effect/alien/resin/sticky/thin
-	name = "thin sticky resin"
-	desc = "A thin layer of disgusting sticky slime."
-	max_integrity = 6
-	slow_amt = 4
-
-	ignore_weed_destruction = FALSE
-
+	H.next_move_slowdown += 6
```
Comment (BraveMole) [inline code/game/objects/structures/xeno.dm:108]: buffed because it's a temp structure now

**Comment by GradusNL** [comment]
Seems like too many different kinds of weeds to me. The weeds themselves don't seem to be thought out that well either.
- Why split plasma and health regen? Forcing you to rest on two different kinds of weeds to regain them both seems like a **massive** nerf, even if they individually regen faster. 
- Only speedy weed increasing speed means that they are basically required since xeno's are very slow off weeds. This will probably invalidate the sticky and toxin weeds. Being slow in front of marines means death, so you will want speedy on the frontline too.
- No clue why you want to add both sticky and toxin weeds, since they are used in the same situation. One or the other will become the meta, making the other useless. I don't like the idea of damage dealing weeds either. For that you've got plenty of other traps.

File: code/game/objects/effects/weeds.dm [hunk 5]
```diff
@@ -89,7 +89,66 @@
...
+	var/mob/living/carbon/human/H = AM
+
+	if(H.lying_angle)
+		return
+
+	H.next_move_slowdown += 1.5
+
+/obj/effect/alien/weeds/resting
+	name = "resting weeds"
+	desc = "This looks almost confortable."
```
Comment (InterroLouis) [inline code/game/objects/effects/weeds.dm:125]: comfortable

**Comment by EvilMelonTheSecond** [comment]
Instead of having weed types maybe make them like miner upgrades (Either with limited supply and higher effects or infinite supply and limited effects). I think it would be more intuitive to build weed types after you are done weeding an area.

**Comment by Surrealaser** [comment]
Also the other big issue: weed wars, and anti-QoL as builder castes bitch at each other and have to deal with other people's wrongweeds in the wrong places. Maybe this shit settles as people become comfortable with the mechanic but I can see it being a real problem.

**Comment by usnpeepoo** [comment]
way too many weeds to be viable, maybe tone it down to 2 or 3

**Comment by BraveMole** [comment]
> The weed types don't seem to be thought out that well.
> 
> * Why split plasma and health regen? Forcing you to rest on two different kinds of weeds to regain them both seems like a **massive** nerf, even if they individually regen faster.

Faire point, i merged the two

> * Only speedy weed increasing speed means that they are basically required since xeno's are very slow off weeds. This will probably invalidate the sticky and toxin weeds. Being slow in front of marines means death, so you will want speedy on the frontline too.

> * No clue why you want to add both sticky and toxin weeds, since they are used in the same situation. One or the other will become the meta, making the other useless. I don't like the idea of damage dealing weeds either. For that you've got plenty of other traps.
> 
> Here's how I predict this is going to pan out. You'll get a combo of plasma and health regen in cover spots across the map and the rest is weeded with speedy weeds. Sticky/toxin will only be used behind the marines to block a retreat, since on frontline use it will slow down xeno's just as much as marines. You can't change weed types either, so that means you can't change sticky to speedy when marines get pushed back.

You can change the type of weeds (by killing the middle node, which takes 0.2 sec). And I'm not sure what you are saying is an issue, that means xeno will use different weeds in different scenarios.

There will probably be a meta installed farily quickly on how to place weeds, but that's how things are everywhere. On this own, this PR won't give much creativity for xeno once the meta is intalled, but we can capitalise on this PR to add more content later.

> Also, are you going to remove basic/regular weed or is this in addition to? Because if it is in addition to and regular weed stays the same I would just always use regular weed for the reasons I gave before. The small bonuses the new weed types give are in no way enough to justify the lack of all other attributes regular weed has.

Yeah, these specs weeds will be the only type of weeds available

**Comment by BraveMole** [comment]
> Instead of having weed types maybe make them like miner upgrades (Either with limited supply and higher effects or infinite supply and limited effects). I think it would be more intuitive to build weed types after you are done weeding an area.

So you build basic weeds, and then you upgrade them? Why not, i'll consider this if this PR turns out to be not good

**Comment by GradusNL** [comment]
> You can change the type of weeds (by killing the middle node, which takes 0.2 sec). And I'm not sure what you are saying is an issue, that means xeno will use different weeds in different scenarios.

My point is that they won't really use different weeds. Speedy is mandatory, since the game is designed around xeno's being on weeds. The off-weed speed of xeno's is absolutely garbage compared to marines. In order to neutralize the speed advantage of marines off-weed you either need to slow them down or speed yourself up. Good marines won't advance across sticky weed (i.e. no effect) so your best option is speedy weed, since you can control where you yourself move. Both sticky and toxin are dead on arrival because they do nothing to remove the speed advantage. The only reason to use resting weed is because they are mandatory to get any recovery, much like how the only reason to place silo's is because they are mandatory to avoid hive death.

tl;dr half of your pr is dead on arrival because you are removing the speed bonus which xeno's are designed around from most weed types.

**Comment by BraveMole** [comment]
@GradusNL You are waaaay to much affirmative, you literraly can't really know for sure what will happen.

Other people are saying that toxin will be the meta, other than it will be sticky, etc...

File: code/modules/mob/living/carbon/xenomorph/life.dm [hunk 6]
```diff
@@ -159,6 +161,7 @@
 	if(lying_angle || resting)
 		plasma_gain *= 2  // Doubled for resting
 
+	plasma_gain *= weeds.color_variant == RESTING_COLOR ? 1.2 : 1
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/life.dm:164]: Why is this not just a weeds var

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/controllers/subsystem/weeds_decay.dm [hunk 7]
```diff
@@ -30,11 +30,11 @@ SUBSYSTEM_DEF(weeds_decay)
 		for(var/direction in GLOB.cardinals)
 			var/turf/adj = get_step(T, direction)
 			if(locate(/obj/effect/alien/weeds) in adj)
-				decay_chance -= rand(20, 24)
+				decay_chance -= rand(15, 20)
 
 		if(prob(decay_chance))
 			W.parent_node = null // So it wont try to regrow
-			addtimer(CALLBACK(GLOBAL_PROC, .proc/qdel, W), rand(1, 7 SECONDS))
+			addtimer(CALLBACK(GLOBAL_PROC, .proc/qdel, W), rand(1, 3 SECONDS))
```
[the remark this answers is not in the document]
Comment (BraveMole) [reply code/controllers/subsystem/weeds_decay.dm:37]: i'm buffing weed decays, because they become obnoxious for marines (toxin and resin weeds)

File: code/game/objects/effects/weeds.dm [hunk 8]
```diff
@@ -86,7 +86,58 @@
...
+
+	if(CHECK_MULTIPLE_BITFIELDS(AM.flags_pass, HOVERING))
+		return
+
+	var/mob/living/carbon/human/H = AM
+
+	if(H.lying_angle)
+		return
+
+	H.apply_damage(2, TOX)
```
[the remark this answers is not in the document]
Comment (BraveMole) [reply code/game/objects/effects/weeds.dm:140]: Ok

File: code/game/objects/effects/weeds.dm [hunk 9]
```diff
@@ -138,18 +188,29 @@
...
 	var/node_turfs = list() // list of all potential turfs that we can expand to
+	/// What type of weeds this node spreads
+	var/weed_type = /obj/effect/alien/weeds
 
 /obj/effect/alien/weeds/node/Destroy()
 	. = ..()
 	SSweeds_decay.decay_weeds(node_turfs)
 
+/obj/effect/alien/weeds/node/attack_alien(mob/living/carbon/xenomorph/X, damage_amount, damage_type, damage_flag, effects, armor_penetration, isrightclick)
+	if(X.status_flags & INCORPOREAL)
```
[the remark this answers is not in the document]
Comment (BraveMole) [reply code/game/objects/effects/weeds.dm:202]: ok

File: code/game/objects/effects/weeds.dm [hunk 10]
```diff
@@ -138,18 +188,29 @@
...
+	X.visible_message("<span class='danger'>\The [X] removed \the [src]!</span>", \
+		"<span class='danger'>We remove \the [src]!</span>", null, 5)
+	take_damage(max_integrity)
+	return TRUE
+
 
 /obj/effect/alien/weeds/node/update_icon()
+	. = ..()
 	overlays.Cut()
 	overlays += node_icon
```
[the remark this answers is not in the document]
Comment (BraveMole) [reply code/game/objects/effects/weeds.dm:215]: ok

File: code/game/objects/structures/xeno.dm [hunk 11]
```diff
@@ -84,10 +84,12 @@
 	max_integrity = 36
 	layer = RESIN_STRUCTURE_LAYER
 	hit_sound = "alien_resin_move"
-	var/slow_amt = 8
-
 	ignore_weed_destruction = TRUE
 
+/obj/effect/alien/resin/sticky/Initialize()
+	. = ..()
+	QDEL_IN(src, 3 MINUTES)
```
[the remark this answers is not in the document]
Comment (BraveMole) [reply code/game/objects/structures/xeno.dm:91]: Normal sticky is gone (can no longer be built by hivelord, queen etc.)
Only the resin built by sticky spit remains, and sticky + slow weeds will be shit for marines.

So xeno can put up temproray sticky by spiting, but fuck mazes made off sitcky, that are banes to build, and banes to destroy

File: code/modules/mob/living/carbon/xenomorph/abilities.dm [hunk 12]
```diff
@@ -314,8 +331,8 @@
 		new_resin = new X.selected_resin(T)
 
 	switch(X.selected_resin)
-		if(/obj/effect/alien/resin/sticky)
-			plasma_cost = initial(plasma_cost) / 3
+		if(/obj/effect/alien/weeds/node/sticky)
+			plasma_cost = initial(plasma_cost) * 10
```
[the remark this answers is not in the document]
Comment (BraveMole) [reply code/modules/mob/living/carbon/xenomorph/abilities.dm:335]: ah, should remove

File: code/__DEFINES/_math.dm [hunk 13]
```diff
@@ -7,7 +7,8 @@
 
 #define CARDINAL_DIRS list(1,2,4,8)
 #define CARDINAL_ALL_DIRS list(1,2,4,5,6,8,9,10)
-#define cheap_hypotenuse(Ax,Ay,Bx,By) (sqrt(abs(Ax - Bx)**2 + abs(Ay - By)**2)) //A squared + B squared = C squared
```
[the remark this answers is not in the document]
Comment (BraveMole) [reply code/__DEFINES/_math.dm:10]: wrong, useless, absolutly not cheap, shit math, not used

File: code/modules/mob/living/carbon/xenomorph/life.dm (hunk 6 above)
Comment (BraveMole) [reply code/modules/mob/living/carbon/xenomorph/life.dm:164]: I don't see why it should

File: code/__DEFINES/xeno.dm [hunk 14]
```diff
@@ -1,2 +1,39 @@
...
+#define STICKY_WEED "sticky weed sac"
+#define RESTING_WEED "resting weed sac"
+#define TOXIN_WEED "toxin weed sac"
+
+//List of weed types
+GLOBAL_LIST_INIT(weed_type_list, list(
+		/obj/effect/alien/weeds/node,
+		/obj/effect/alien/weeds/node/sticky,
+		/obj/effect/alien/weeds/node/resting,
+		/obj/effect/alien/weeds/node/toxin,
```
[the remark this answers is not in the document]
Comment (BraveMole) [reply code/__DEFINES/xeno.dm]: Still not a typepath, because /obj/effect/alien/weeds/node should not be available (it's just a theoritical weed)

File: code/modules/mob/living/carbon/xenomorph/life.dm (hunk 6 above)
Comment (TiviPlus) [reply code/modules/mob/living/carbon/xenomorph/life.dm:164]: Because it means that A) the behavior is generic B) no magic numbers

File: code/modules/mob/living/carbon/xenomorph/life.dm [hunk 15]
```diff
@@ -66,12 +66,12 @@
...
-
-	if(locate(/obj/effect/alien/weeds) in T || xeno_caste.caste_flags & CASTE_INNATE_HEALING) //We regenerate on weeds or can on our own.
+	var/obj/effect/alien/weeds/weed = locate() in T
+	if(weed || xeno_caste.caste_flags & CASTE_INNATE_HEALING) //We regenerate on weeds or can on our own.
 		if(lying_angle || resting || xeno_caste.caste_flags & CASTE_QUICK_HEAL_STANDING)
-			heal_wounds(XENO_RESTING_HEAL * ruler_healing_penalty, TRUE)
+			heal_wounds(XENO_RESTING_HEAL * ruler_healing_penalty * (weed?.color_variant == RESTING_COLOR ? 1.2 : 1), TRUE)
 		else
-			heal_wounds(XENO_STANDING_HEAL * ruler_healing_penalty, TRUE) //Major healing nerf if standing.
+			heal_wounds(XENO_STANDING_HEAL * ruler_healing_penalty * (weed?.color_variant == RESTING_COLOR ? 1.2 : 1), TRUE) //Major healing nerf if standing.
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/life.dm:74]: See this, ditto

File: code/game/objects/effects/weeds.dm [hunk 16]
```diff
@@ -163,16 +234,35 @@
 	update_icon()
 
 	// Generate our full graph before adding to SSweeds
-	node_turfs = filled_turfs(src, node_range, "square")
+	node_turfs = filled_turfs(src, 2, "square")
```
Comment (TiviPlus) [inline code/game/objects/effects/weeds.dm:237]: Magic number what is this

Comment (TiviPlus) [reply code/game/objects/effects/weeds.dm:237]: Also where did node range go

File: code/game/objects/effects/weeds.dm [hunk 17]
```diff
@@ -86,7 +86,65 @@
...
+
+	var/mob/living/carbon/human/victim = AM
+
+	if(victim.lying_angle)
+		return
+
+	victim.apply_damage(3, BRUTE)
+
+	if(prob(15))
+		victim.emote("pain")
```
Comment (TiviPlus) [inline code/game/objects/effects/weeds.dm:147]: Add a chat message

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/abilities.dm [hunk 18]
```diff
@@ -136,13 +138,32 @@
...
+
+/datum/action/xeno_action/plant_weeds/proc/choose_weed()
+	var/weed_choice = show_radial_menu(owner, owner, GLOB.weed_images_list, radius = 48)
+	if(!weed_choice)
+		return
+	for(var/obj/effect/alien/weeds/node/weed_type_possible AS in GLOB.weed_type_list)
+		if(initial(weed_type_possible.name) == weed_choice)
+			weed_type = weed_type_possible
+			break
+	to_chat(owner, "<span class='notice'>We will now spawn <b>[weed_choice]\s</b> when using the spawn hugger ability.</span>")
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/abilities.dm:159]: 🍝

File: code/game/objects/effects/weeds.dm [hunk 19]
```diff
@@ -26,10 +33,10 @@
 
 	if(!isnull(node))
 		if(!istype(node))
-			CRASH("Weed craeted with non-weed node. Type: [node.type]")
+			CRASH("Weed created with non-weed node. Type: [node.type]")
 		parent_node = node
-
-	update_sprite()
+		color_variant = node.color_variant
+	update_icon()
```
Comment (TiviPlus) [inline code/game/objects/effects/weeds.dm:39]: does resting_buff get udated as well

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/abilities.dm [hunk 20]
```diff
@@ -136,13 +138,32 @@
...
+	new weed_type(owner.loc)
 	playsound(owner.loc, "alien_resin_build", 25)
 	GLOB.round_statistics.weeds_planted++
 	SSblackbox.record_feedback("tally", "round_statistics", 1, "weeds_planted")
 	add_cooldown()
 	return succeed_activate()
 
+/datum/action/xeno_action/plant_weeds/alternate_keybind_action()
+	INVOKE_ASYNC(src, .proc/choose_weed)
+
```
[the remark this answers is not in the document]
Comment (BraveMole) [reply code/modules/mob/living/carbon/xenomorph/abilities.dm:150]: radial wheels have sleep in it i think

File: code/game/objects/effects/weeds.dm (hunk 19 above)
Comment (BraveMole) [reply code/game/objects/effects/weeds.dm:39]: no, good catch

**Review by Kuro020 — APPROVED** [review]
They don't make my eyes burn so it's ok

File: code/game/objects/effects/weeds.dm [hunk 21]
```diff
@@ -86,7 +87,67 @@
 		icon_state = "base"
 	else
 		icon_state = "weed_dir[my_dir]"
+	icon_state += color_variant
+
+/obj/effect/alien/weeds/speed
+	name = "speed weeds"
+	desc = "A layer of oozy slime"
```
Comment (XSlayer300) [inline code/game/objects/effects/weeds.dm:94]:
```suggestion
	desc = "A layer of oozy slime, it feels slick, but not as slick for you to slip."
```

File: code/game/objects/effects/weeds.dm [hunk 22]
```diff
@@ -86,7 +87,67 @@
...
+
+/obj/effect/alien/weeds/speed/Crossed(atom/movable/AM)
+	. = ..()
+	if(isxeno(AM))
+		var/mob/living/carbon/xenomorph/X = AM
+		X.next_move_slowdown += X.xeno_caste.weeds_speed_mod
+
+/obj/effect/alien/weeds/sticky
+	name = "sticky weeds"
+	desc = "A layer of disgusting sticky slime."
```
Comment (XSlayer300) [inline code/game/objects/effects/weeds.dm:104]:
```suggestion
	desc = "A layer of disgusting sticky slime, it feels like it's going to slow your movement down."
```

File: code/game/objects/effects/weeds.dm [hunk 23]
```diff
@@ -86,7 +87,67 @@
...
+	var/mob/living/carbon/human/victim = AM
+
+	if(victim.lying_angle)
+		return
+
+	victim.next_move_slowdown += 1.5
+
+/obj/effect/alien/weeds/resting
+	name = "resting weeds"
+	desc = "This looks almost comfortable."
```
Comment (XSlayer300) [inline code/game/objects/effects/weeds.dm:124]:
```suggestion
	desc = "This looks almost comfortable, while not as special for you, it is certainly special for the otherwordly creatues out from the dark, as if they regain their strength faster."
```

File: code/game/objects/effects/weeds.dm [hunk 24]
```diff
@@ -86,7 +87,67 @@
...
+	color_variant = RESTING_COLOR
+	resting_buff = 1.2
+
+/obj/effect/alien/weeds/barbed
+	name = "barbed weeds"
+	desc = "This looks pointy"
```
Comment (XSlayer300) [inline code/game/objects/effects/weeds.dm:130]:
```suggestion
	desc = "This looks very sharp and pointy, as if thorns poke you from all sides. Best not to step on it with your human limbs."
```

File: code/game/objects/effects/weeds.dm [hunk 25]
```diff
@@ -86,7 +87,67 @@
...
+
+	var/mob/living/carbon/human/victim = AM
+
+	if(victim.lying_angle)
+		return
+
+	victim.apply_damage(1.5, BRUTE)
+
+	if(prob(15))
+		to_chat(victim, span_warning("You hurt yourself on the barbed weeds"))
```
Comment (XSlayer300) [inline code/game/objects/effects/weeds.dm:149]:
```suggestion
		to_chat(victim, span_warning("You are cut from the barbed weeds."))
```

**Comment by Pariah919** [comment]
Barbed weeds are awful to play agains tsince they can be spammed and deal like 1 brute per step which is annoying to deal with since you need to be premededded
in general i don't like barbed weeds as a game mechanic and it's awful to deal with since one brute adds up, i was literally never at full hp and this just makes dealign with mazes worse

**Comment by jroinc** [comment]
Doesn't actually have a non-fiddle-with-code way to use, and is the only xeno ability without a HUD button or verb associated with it.

-1, shit implementation.

**Comment by BraveMole** [comment]
> non-fiddle-with-code

A keybind? wow

**Comment by jroinc** [comment]
> 
> 
> > non-fiddle-with-code
> 
> A keybind? wow. You know that 90% of the games out there works with only keybinds? You gonna say it's shit implementation as well?

You have a speshul snowflake xeno ability that doesn't function the way all others do. Please add a button or a verb. There's no reason not to do that besides "I want mine to be speshul".

**Comment by Hekzder** [comment]
I really like this PR in concept but it could definitely use some work. 

1. I agree full heartedly with @jroinc that this should not be the only Xeno ability in the game that is key bind only, some kind of functionality so that you can just click it to change weeds would not only make sense and match every other ability in the game, but make it less confusing for newer or less keybind inclined people to use.

2. I also generally agree with @Pariah919 about barbed weeds as a game mechanic. In practice I don't actually think they are that good as of when I tested them, contrary to what he says, but fundamentally I think "damaging weeds" is a bad idea. Weeds should be utility based and IMO generally around buffing up Xenos or maybe _debuffing_ Marines, but not straight up damaging them. Even with the debuffing part, I think weeds should primarily be focused just around effects on Xenos.

3. When I was last playing, sticky weeds were basically obsolete because barbed weeds dealing damage + pain was just better. I see that sticky weeds have since been buffed but so have barbed weeds, kind of bringing me to my last point that there shouldn't be a ton of Marine inhibiting weeds. Probably just one at most, likely with the sticky effect of slowing down, having multiple Marine inhibiting weeds not only makes things kind of messy but is honestly probably much harder to balance. Just makes things easier all around to only have one.

Overall, like I said, I really like the PR, and in practice the weeds aren't too terrible when I was using them. Using the healing weeds for little healing spots is great and neat to design around, barbed were okay at specific chokepoints and being able to conceal the node behind something. But speed still generally dominated, though I feel that is partially just due to people being used to it, and making Xenos faster will _always_ be good, whether it is frontline or backline weeds.

**Comment by TiviPlus** [comment]
@BraveMole is this ready again

**Comment by SplinterGP** [comment]
Should possibly work in a way that you can click to change as well alongside keybind, theres other abilites that do something like that i believe

**Comment by BraveMole** [comment]
> @BraveMole is this ready again
@TiviPlus yes

**Comment by Al-1ce** [comment]
This is a great idea but I have a bad feeling that in practice this'll just make things more difficult.

**Comment by rusher210** [comment]
i like these new weeds, they open up alternatives for hive building (even if they'll get eventually OBd out of existance after 20 painful minutes of building and boredom)
cutting the amount of weeds to 3 is a good move IMO

**Comment by lKiseki** [comment]
Barbed weeds have innate slowdown.
Why. Why. Why. Why. No. That's a really bad idea.

**Comment by davidchan13** [comment]
Removing barbed weeds is a bit of a knee jerk reaction, IMO. Barbed Weeds slow down was perhaps a step too far, when you consider that sticky resin can be placed over a defensive area. But the rest of the concept was good and allowed Xenos to punish overzealous marines pushing into territory still controlled by xenos trying to secure a kill or failing to clear resin when pushing.

Speed weeds just don't offer anything interesting to be used. Tunnels allow xenos move across the map, and sticky resin/sticky weeds will just slow down marines more than speed improves reactions of xenos.

Consider adding the ability for xenos to destroy/remove/replace unwanted weed nodes. As the state of a battle changes, xenos may want to reconfigure their weed layouts and change the type of weeds in an area, and currently lack the ability to do so without spending a lot of plasma to try and overwrite non-node weeds with nodes.

**Comment by BraveMole** [comment]
> Removing barbed weeds is a bit of a knee jerk reaction, IMO. Barbed Weeds slow down was perhaps a step too far, when you consider that sticky resin can be placed over a defensive area. But the rest of the concept was good and allowed Xenos to punish overzealous marines pushing into territory still controlled by xenos trying to secure a kill or failing to clear resin when pushing.

Eh, people hated it, i won't push for that

> Speed weeds just don't offer anything interesting to be used. Tunnels allow xenos move across the map, and sticky resin/sticky weeds will just slow down marines more than speed improves reactions of xenos.

I kinda disagree, both have advantages. Speed weeds are excellent for runner like classes, because wasd is really improved

> Consider adding the ability for xenos to destroy/remove/replace unwanted weed nodes. As the state of a battle changes, xenos may want to reconfigure their weed layouts and change the type of weeds in an area, and currently lack the ability to do so without spending a lot of plasma to try and overwrite non-node weeds with nodes.

It exist. You can attack a node, or just place another node on top of it

File: code/game/objects/effects/weeds.dm [hunk 26]
```diff
@@ -138,41 +167,122 @@
...
+	element_flags = ELEMENT_BESPOKE
+	id_arg_index = 2
+	///How much it slow down on crossed
+	var/slow_amount = 0
+
+/datum/element/slowing_on_crossed/Attach(datum/target, slow_amount)
+	. = ..()
+	src.slow_amount = slow_amount
+	RegisterSignal(target, COMSIG_MOVABLE_CROSSED_BY, .proc/slow_down_crosser)
+
```
Comment (TiviPlus) [inline code/game/objects/effects/weeds.dm:259]: We have an elements folder for a reason

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Comment by polyakovGit** [comment]
grass cells do not always connect
Screenshot_1

**Comment by Arcetro** [comment]
I like this PR. Just a stupid comment to add. what about if the weed absorbs blood? Just thinking out loud.
